storage/innobase/include/srw_lock.h

```diff
@@ -261,7 +261,7 @@ class ssux_lock_impl
     return false;
   }
 
-  void rd_lock() noexcept { if (!rd_lock_try()) rd_wait(); }
+  void rd_lock() noexcept { if (!spinloop || !rd_lock_try()) rd_wait(); }
```

Comment:
Note: On an Intel Xeon system where I tested, this change (as well as the corresponding adjustment to `rd_wait()`) shrunk the `CMAKE_BUILD_TYPE=RelWithDebInfo` `sql/mariadbd` executable size by about 30 KiB. The `rd_lock_try()` would translate into a loop around `lock cmpxchg`. I didn’t observe a significant performance difference; this patch could slightly improve the performance.